src/rulesets/oas/__tests__/__snapshots__/parameter-description.ts.snap

```diff
@@ -0,0 +1,87 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`parameter-description return errors if operation level parameter description is missing 1`] = `
+Array [
```

Comment:
Snapshots are a nightmare in Spectral.
When I was changing the diagnostics I had to update about ~200 (or more) snapshots...
I know all the other rules have similar assertions, so I don't oblige you to change anything, but I just wanted to point it out.

Reply:
Good point but I don't think it has anything to do with snapshot technically. If we didn't have snapshots you'd need to update 200 tests. I'm not sure why but I'm guessing that's because we are not mocking stuff in unit tests (or write too many integration tests?)

Reply:
Maybe maybe not. Snapshots contain everything, where as assertions only contain what you care about asserting. If an arbitrary property changes in 200 snapshots that sucks, but you might not have had to update 200 tests.

See other comment about snapshots. 
1. We don’t want to hide our assertions from the test file it makes it hard to read 
2. We don’t want our tests relying on a file which can change and accidentally get committed in a huge PR leading to false positives

Let’s just use expects :)

Reply:
Right, it looks like it's a very delicate subject matter and mostly subjective ;)

I'm on the edge with snapshots personally. I try to avoid them to (like you said) increase visibility, but at the same time I disagree that they 'assert too much'. I actually think the opposite: 'expects.equalsTo' asserts too little.

E.g. 

>  If an arbitrary property changes in 200 snapshots that sucks,

You say it sucks, I say it's a bug that we missed to cover ;)

Anyway, I will see if I can reasonably ditch some of the snaphots but I'd like us to have a proper discussion about this - we need clear guidelines otherwise we'll keep arguing  about this in every PR ;)

Reply:
A bug that we missed to cover? No, im saying that if a test cares about only part of the response, and a completely unrelated property changes, then that is not something which should be causing a failure.

An assertion should only be asserting what it cares about. Instead of saying "I am interested in seeing that A, B and C are included in this object" you say "I can about this object matching a snapshop" and that giant snapshot has "A, B, C, D, E, F, G, H, I, 345, Tuesday, Ice Cream...." then you have a test that is 1.) unreadable, 2.) vague to the point of being confusing.

You can use https://jestjs.io/docs/en/expect.html#expectobjectcontainingobject to make awesome readable tests that are clear what they are asserting. MatchSnapshot is a nuclear option. 

We will make style guides one day, but for now I am just seeing a lot of people saying please stop using them, so that doesn't seem like much of an issue, lets just stop using them. :)

Reply:
Me and @P0lip, during a personal conversation on Slack, came out with the perfect implementation of snapshots:

![carbon](https://user-images.githubusercontent.com/1416224/56372140-b6ebb980-61fe-11e9-89a5-d4854c4299b5.png)

Reply:
I'd say snapshots are only fine if the content is not meant to change, i.e. you have a parser, babel plugin etc. You usually don't expect the content to change much + everything is very important in such case + it's almost impossible (well, it's just unreadable) to use some regular assertion.

We def should come up with some guideline for that as indeed these discussions take place quite often and we've never reached any agreement.

Reply:
Cool, I will take a look and minimize snaphots ;)